### DISCUSSION TITLE
Display Texts in Hexagons

### DISCUSSION THREAD

**Discussion by mertbakir** [marcusolsson/grafana-hexmap-panel#18] [Ideas]
As we already discussed in [this](https://github.com/marcusolsson/grafana-hexmap-panel/issues/2) thread, it would be useful to display labels or metrics and even emojis inside the hexagons.

**Comment by marcusolsson** [comment]
Seems like there's some more interest for this. I'll look at adding this during the week.

**Comment by phillebaba** [comment]
Just want to start out with saying this project is awesome, and has solved some major pain points for me :smiley: 

I would also be interested in this feature. Having the option to display a field value inside of a hexagon would make it easier to understand the data when the panel is displayed in kiosk mode. Right now I need to open the dashboard on my computer and hover over the cell to get this information.

I have done very little Grafana plugin work so don't really know who difficult this would be to add.

**Comment by marcusolsson** [comment]
@phillebaba Thanks! ❤️ If you know your way around TypeScript (but new to plugins), and would like to try contributing, I'd be happy to walk you through the plugin. If you're interested, we could hop into a call ([Calendly](https://calendly.com/marcusolsson)).

**Comment by MattIPv4** [comment]
👋 Would just like to add a +1 to this feature request -- it seems like this plugin is the best alternative to polystat for table data, but not having anything displayed in the tiles is a bummer.

**Comment by phillebaba** [comment]
I took a stab at it and got a simple solution working, I can create a draft PR this weekend. My major blocker right now is figuring out SVG text sizing so that it fits within the polygon. Does anyone have any pointers for how this could be solved?

**Comment by marcusolsson** [comment]
Nice! I had a go at adaptive text sizing in another plugin, but haven't merged it yet since it does seem to come with a pretty hefty performance hit. I imagine it would be even worse for this plugin.

Maybe check it out and see if you can figure out if there's anything we can improve it?

https://github.com/marcusolsson/grafana-treemap-panel/pull/13

**Comment by mvadu** [comment]
> I took a stab at it and got a simple solution working, I can create a draft PR this weekend. My major blocker right now is figuring out SVG text sizing so that it fits within the polygon. Does anyone have any pointers for how this could be solved?

Check the stat panel's text rendering (i think this one https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/stat/StatPanel.tsx) it does shrink when the available panel area shrinks.

**Comment by phillebaba** [comment]
Sorry for the late update. My computer self destructed and I had not pushed a branch before it happened. So I ended up rage quitting on the whole situation. Now I am back at square one and will have to make a second attempt...
